proto/buf/alpha/registry/v1alpha1/image.proto

```diff
@@ -25,10 +25,79 @@ service ImageService {
   rpc GetImage(GetImageRequest) returns (GetImageResponse);
 }
 
+// ImageMask is used in GetImageRequest to specify which parts of an image are
+// requested.
+enum ImageMask {
+  IMAGE_MASK_UNSPECIFIED = 0;
+
+  // IMAGE_MASK_MESSAGES refers to ImageFile's `google.protobuf.DescriptorProto
+  // message_type` field.
+  IMAGE_MASK_MESSAGES = 1;
+
+  // IMAGE_MASK_ENUMS refers to ImageFile's `google.protobuf.EnumDescriptorProto
+  // enum_type` field.
+  IMAGE_MASK_ENUMS = 2;
+
+  // IMAGE_MASK_SERVICES refers to ImageFile's
+  // `google.protobuf.ServiceDescriptorProto service` field.
+  IMAGE_MASK_SERVICES = 3;
+}
+
 message GetImageRequest {
   string owner = 1;
   string repository = 2;
   string reference = 3;
+
+  // Exclude files from imported buf modules in this image.
+  bool exclude_imports = 4;
+
+  // Exclude source_code_info fields from each ImageFile.
+  bool exclude_source_info = 5;
+
+  // When specified the returned image will only contain the necessary files to
+  // describe these types. Accepts messages, enums and services. All types must
+  // be defined in the buf module, types in dependencies are not accepted.
+  repeated string types = 6;
+
+  // When `exclude_extensions` is true, images may omit extension declarations
+  // for regular proto2 messages. When used in combination with `types`,
+  // extensions will not be considered for determining what types are referenced
+  // by the requested type.
+  //
+  // For example, in the following example proto structure:
+  //
+  //    --- foo.proto ---
+  //    message Foo {
+  //      optional Bar bar = 1 [(my_option).field = "buf"];
+  //      extensions 2 to 3;
+  //    }
+  //    --- baz.proto ---
+  //    extend Foo {
+  //      optional Baz baz = 2;
+  //    }
+  //    --- qux.proto ---
+  //    extend google.protobuf.FieldOptions {
+  //      optional Qux my_option = 51234;
+  //    }
+  //
+  //  When `exclude_extensions` is true and `types` is "Foo", the resulting image
+  //  will contains types for "Foo", "Bar" and "Qux", but not "Baz".
+  bool exclude_extensions = 7;
+
+  // When `exclude_extensions` is true, images may omit extends declarations for
```

Comment:
Any option applied is a part of that definition, i.e. a field option is a part of that definition. Just as you wouldn't say "exclude any `deprecated` options", you can't then say that custom field options are ignored. Again unless I'm missing something, this option shouldn't exist.